### ISSUE TITLE
Proposal: Update the schema of mps-data to more closely conform the structure in SNPedia

### ISSUE THREAD

**Issue by livtanong** [snpeek/snpeek.github.io#17]
Currently, `mps-data.json` is a dictionary, mapping SNP reference IDs to a set of fields: [phenotype, pathogenic, gene, notes]

My understanding is that in reality, each SNP is a "folder", and the smallest unit of information here is the genotype. It is the genotype that has a phenotype (or possibly multiple?), roughly corresponding to the Summary in SNPedia. It's also the genotype that can describe the degree to which it is pathogenic or beneficial, corresponding to the Magnitude (and color) section of its entry in SNPedia.

My proposal for the new schema therefore is:

```
interface SnpGenotype {
  alleles: string // By my understanding, CT == TC, so we can enforce a standard.
  magnitude: number // positive numbers are good, negative numbers are pathogenic
  phenotypes: string[] // basically like tags, since I noticed some genotypes have multiple effects
  notes: string // To add context, in case a phenotype has very conditional expression
}
interface Snp {
  rsid: string
  genotypes: SnpGenotype[]
  gene: string
  notes: string // just in case.
}
type MpsData = Map<string, Snp>
```

Of course, this will require a rework of how analysis happens (likewise, for rendering the table), but it shouldn't be difficult at all.

**Comment by d6e** [comment]
Yeah, that makes sense to me, I've been pretty busy lately and I dont' have the spare cycles but feel free to push an MR for this!

**Comment by livtanong** [comment]
Gotcha! I'll work on this next.

**Comment by livtanong** [comment]
So it turns out, Magnitude (and by extension, Max Magnitude) are purely SNPedia things, and entirely subjective.
https://www.snpedia.com/index.php/Magnitude

I believe we're at a crossroads. Is SNPeek meant to be tied to SNPedia, or independent of it? I would be more confident of sticking to the SNPedia shape, but there are a lot of SNPs that don't show up in SNPedia. That said, we can also base it off of SNPedia, but work in the fact that we don't have perfect information, and we can just make some of the fields nullable.

There also exists a middle ground, where we come up with our *own* magnitude scale, but base it initially off of SNPedia. And through reading the literature, come up with our own summaries (and magnitude and phenotype) of the studies associated with the particular SNP.

**Comment by d6e** [comment]
> There also exists a middle ground, where we come up with our own magnitude scale, but base it initially off of SNPedia. And through reading the literature, come up with our own summaries (and magnitude and phenotype) of the studies associated with the particular SNP.

Yeah, that's essentially the assumption we've been working under. The whole magnitude thing is kind of tricky and subjective

**Comment by livtanong** [comment]
We probably need to move away from SNPedia, but because we're already talking about the schema for `mps-data.json`, I'm noting with this comment that this discussion is somewhat related to #24
